src/Integration/FrameContextifierIntegration.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Integration;
+
+use Psr\Log\LoggerInterface;
+use Sentry\Event;
+use Sentry\SentrySdk;
+use Sentry\Stacktrace;
+use Sentry\State\Scope;
+
+/**
+ * This integration reads excerpts of code around the line that originated an
+ * error.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+final class FrameContextifierIntegration implements IntegrationInterface
+{
+    /**
+     * @var int The maximum number of lines of code to read
+     */
+    private $contextLines;
+
+    /**
+     * @var LoggerInterface|null A PSR-3 logger
+     */
+    private $logger;
+
+    /**
+     * Creates a new instance of this integration.
+     *
+     * @param int             $contextLines The maximum number of lines of code to read
+     * @param LoggerInterface $logger       A PSR-3 logger
+     */
+    public function __construct(int $contextLines = 5, ?LoggerInterface $logger = null)
+    {
+        if ($contextLines < 0) {
+            throw new \InvalidArgumentException(sprintf('The value of the $maxLinesToFetch argument must be greater than or equal to 0. Got: "%d".', $contextLines));
+        }
+
+        $this->contextLines = $contextLines;
+        $this->logger = $logger;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setupOnce(): void
+    {
+        Scope::addGlobalEventProcessor(static function (Event $event): Event {
+            $client = SentrySdk::getCurrentHub()->getClient();
+
+            // Avoid doing double work if the deprecated `context_lines` option
+            // is still in use. Since the option can change at runtime, it's safer
+            // to check its value here every time rather than when deciding whether
+            // the integration should be added or not to the client
+            if (null === $client || null !== $client->getOptions()->getContextLines(false)) {
+                return $event;
+            }
+
+            $integration = $client->getIntegration(self::class);
+
+            if (null === $integration) {
+                return $event;
+            }
+
+            if (null !== $event->getStacktrace()) {
+                $integration->addContextToStacktraceFrames($event->getStacktrace());
+            }
+
+            foreach ($event->getExceptions() as $exception) {
+                if (!isset($exception['stacktrace'])) {
+                    continue;
+                }
+
+                $integration->addContextToStacktraceFrames($exception['stacktrace']);
```

Comment:
Should you not check the type before passing it in?

Reply:
It's not worth it because:

- I plan to phase out all these arrays with value objects instead to solve the issue upstream
- If the type is wrong PHP will complain due to the typehinting of the parameter of the function, and imho it's enough. We have to remember that this information is coming from the internals of the event factory, so it's unlikely that is wrong